src/Transport/Amqp/AmqpTransport.php

```diff
@@ -41,13 +41,13 @@ class AmqpTransport
     /**
      * AmqpTransport constructor.
      * @param AmqpConnectionSettings $connectionSettings
-     * @param EventSerializer $serializer
+     * @param MessageSerializer $serializer
      * @param ListenerProvider $listenerProvider
      * @param LoggerInterface $logger
      */
     public function __construct(
         AmqpConnectionSettings $connectionSettings,
-        EventSerializer $serializer,
+        MessageSerializer $serializer,
         ListenerProvider $listenerProvider,
```

Comment:
Once you have added the method `public function messageHasListeners(Message $event): bool`, to the `ListenerProviderInterface`, please use the interface here